### PR TITLE
Version poll

### PR DESCRIPTION
**Pull request by robbrockbank**
## Description
Suggestion that we add a poll option to calicoctl version rather than include a separate calicoctl-wait binary.

This can be used by the calicoctl container to poll indefinitely.

Removes the requirement of the extra binary.

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
@robbrockbank this looks good to me - guessing we need a corresponding docs / manifests change PR?

**Comment by robbrockbank** [comment]
@caseydavenport :  Yup that's correct - didn't want to make the other updates until this was reviewed.  Aim for 3.17 I'm guessing?

File: calicoctl/commands/version.go [hunk 1]
```diff
@@ -17,7 +17,12 @@ package commands
 import (
 	"context"
 	"fmt"
+	"os"
 	"strings"
+	"time"
+
+	"github.com/projectcalico/calicoctl/calicoctl/commands/argutils"
+	v3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
```
Comment (doublek) [inline calicoctl/commands/version.go:25]: nit: Looks like grouping is off.

**Comment by doublek** [comment]
I think there are a few related changes. Off the top of my head:
- docs manifests for running calicoctl as a pod.
- node STs.

File: calicoctl/commands/version.go [hunk 2]
```diff
@@ -35,13 +40,15 @@ func init() {
...
 	doc := `Usage:
-  calicoctl version [--config=<CONFIG>]
+  calicoctl version [--config=<CONFIG>] [--poll=<POLLFREQ>]
 
 Options:
   -h --help             Show this screen.
   -c --config=<CONFIG>  Path to the file containing connection configuration in
                         YAML or JSON format.
                         [default: ` + constants.DefaultConfigPath + `]
+     --poll=<POLL>      Poll for changes to the cluster information at a frequency specified using POLL duration
```
Comment (doublek) [inline calicoctl/commands/version.go:50]: very nit picky: the template `POLLFREQ` and  `POLL` above don't match.

**Comment by caseydavenport** [comment]
> didn't want to make the other updates until this was reviewed. Aim for 3.17 I'm guessing?

Yeah, unless there's a critical reason it should go into v3.16 (but I don't think there is?)

**Pull request merged**